c7n/resources/cloudfront.py

```diff
@@ -181,9 +179,60 @@ def process(self, resources, event=None):
         return results
 
 
+@Distribution.filter_registry.register('mismatch-s3-owner')
+class MismatchS3Owner(Filter):
+    """Check for existence of S3 bucket referenced by Cloudfront,
+       and verify whether owner is different from Cloudfront account owner.
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: mismatch-s3-owner
+                resource: distribution
+                filters:
+                  - type: mismatch-s3-owner
+   """
+
+    schema = type_schema(
+        'mismatch-s3-owner')
+
+    permissions = ('s3:ListBuckets',)
+    retry = staticmethod(get_retry(('Throttling',)))
+
+    def is_s3_domain(self, domain_name):
+        if domain_name.endswith('.s3.amazonaws.com'):
+            return domain_name[:-len('.s3.amazonaws.com')]
+
+        return False
+
+    def process(self, resources, event=None):
+        results = []
+
+        s3_client = local_session(self.manager.session_factory).client(
+            's3', region_name=self.manager.config.region)
+
+        buckets = {b['Name'] for b in s3_client.list_buckets()['Buckets']}
+
+        for r in resources:
+            for x in r['Origins']['Items']:
+                if 'S3OriginConfig' in x:
+                    target_bucket = x['DomainName'].split('.', 1)[0]
+                elif 'CustomOriginConfig' in x and self.is_s3_domain(x['DomainName']):
+                    target_bucket = self.is_s3_domain(x['DomainName'])
+
+                if 'target_bucket' in locals() and target_bucket not in buckets:
```

Comment:
also please set a default value in strong preference to using locals which is highly implicit.